File: app/src/main/java/com/twismart/wallpapershd/ui/main/ListWallpapersContract.java
package com.twismart.wallpapershd.ui.main;

import com.twismart.wallpapershd.data.model.Wallpaper;
import com.twismart.wallpapershd.ui.base.BasePresenter;
import com.twismart.wallpapershd.ui.base.BaseView;

import java.util.ArrayList;

/**
 * Created by sneyd on 5/8/2017.
 **/

public class ListWallpapersContract {

    public interface Presenter<V extends ListWallpapersContract.View> extends BasePresenter<V> {
        void getWallpapersList();
    }

    public interface View extends BaseView {
        void setWallpaperList(ArrayList<Wallpaper> wallpaperList);
    }
}

File: app/src/main/kotlin/com/twismart/wallpapershd/ui/main/ListWallpapersContract.kt
package com.twismart.wallpapershd.ui.main

import com.twismart.wallpapershd.data.model.Wallpaper
import com.twismart.wallpapershd.ui.base.BaseContract

/**
 * Created by sneyd on 5/8/2017.
 **/

class ListWallpapersContract {

    interface Presenter<V : ListWallpapersContract.View> : BaseContract.Presenter<V> {
        fun getWallpapersList()
        fun getAllFavoriteWallpapers()
    }

    interface View : BaseContract.View {
        fun setWallpaperList(wallpaperList: List<Wallpaper>)
    }
}
